client/src/components/subscription/subscriptionForm.js

```diff
@@ -0,0 +1,239 @@
+    event.preventDefault();
+    toggleLoadingState();
+
+    if (frequency !== 'daily' && days.length === 0) {
+      toast.error('Failed to Submit: Dates not selected for reminder.');
+      return;
+    }
+
+    const dates = this.parseDueDate();
+    const dueDate = { frequency, occurence, dates };
+
+    const subscriptionInfo = { name, nickname, reminderDays, amount, dueDate };
+
+    if (method === 'PATCH') {
+      const { prevSubscription } = this.props;
+      subscriptionInfo.subscriptionUuid = prevSubscription.subscriptionUuid;
+    }
+    
+    const subscription = await fetch(`${window.location.pathname}/subscriptions`, {
+      method,
+      headers: { 'Content-type': 'application/json' },
+      body: JSON.stringify(subscriptionInfo)
+    });
+
+    const { status } = subscription;
+    const response = await subscription.json();
+    toggleLoadingState();
+    if (status === 400) {
+      const { errorMessage } = response;
+      toast.error(errorMessage);
+      return;
+    }
+
+    showSubscriptionList();
+    handleSubscriptions(response);
+  }
+
+  handleDays(days) {
+    this.setState({ days });
+  }
+
+  parseDueDate() {
+    const { frequency, occurence, days } = this.state;
+    let parsedDay;
+    switch (frequency) {
+      case 'yearly':
+        parsedDay = days.map((day) => day.toISOString().substring(0, 10));
+        break;
+      case 'monthly':
+        parsedDay = days.map((day) => day.toISOString().substring(0, 10));
+        break;
+      case 'weekly':
+        parsedDay = convertWeekdaysToDates(occurence, days);
+        break;
+      case 'daily':
+        parsedDay = days.map((day) => day.toISOString().substring(0, 10));
+        break;
+      default:
+        break;
+    }
+    return parsedDay;
+  }
+
+  renderSwitch(frequency) {
+    const { occurence, days } = this.state;
+    const weeklyCheckbox = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'].map((day, idx) => 
+      <label htmlFor={day} key={day}>
+        {day}:
+        <input type="checkbox" id={day} name={days} value={idx} onChange={this.handleCheck} checked={days.includes(idx.toString())}/>
+      </label>
+    );
+
+    switch (frequency) {
+      case 'yearly':
+        return (
+          <div>
+            On what day do you want to be reminded?
+            <ReactDayPicker handleUpdate={this.handleDays} />
+          </div>
+        );
+      case 'monthly':
+        return (
+          <div>
+             <label htmlFor="occurence">
+                Every (?) Months:
+                <input type="number" id="occurence" value={occurence} onChange={(event) => this.handleChange(event, 'occurence')} min="0" max="12" />
+              </label>
+              <div>On which day(s) do you want to be reminded?</div>
+              <ReactDayPicker handleUpdate={this.handleDays} canChangeMonth={false} />
+          </div>
+        );
```

Comment:
some indentation fixes here. also for `occurrence` you'll want to replace all instances to make sure spelling is correct

```suggestion
        return (
          <div>
            <label htmlFor="occurence">
              Every (?) Months:
              <input type="number" id="occurrence" value={occurrence} onChange={(event) => this.handleChange(event, 'occurrence')} min="0" max="12" />
            </label>
            <div>On which day(s) do you want to be reminded?</div>
            <ReactDayPicker handleUpdate={this.handleDays} canChangeMonth={false} />
          </div>
        );
```